subprojects/core/src/main/java/org/gradle/internal/operations/logging/LoggingBuildOperationProgressBroadcaster.java

```diff
@@ -61,32 +63,40 @@
     private final OutputEventListenerManager outputEventListenerManager;
     private final BuildOperationListener buildOperationListener;
 
+    @VisibleForTesting
+    OperationIdentifier rootBuildOperation;
+
     public LoggingBuildOperationProgressBroadcaster(OutputEventListenerManager outputEventListenerManager, BuildOperationListener buildOperationListener) {
         this.outputEventListenerManager = outputEventListenerManager;
         this.buildOperationListener = buildOperationListener;
-
         outputEventListenerManager.setListener(this);
     }
 
     @Override
     public void onOutput(OutputEvent event) {
         if (event instanceof RenderableOutputEvent) {
             RenderableOutputEvent renderableOutputEvent = (RenderableOutputEvent) event;
-            if (renderableOutputEvent.getBuildOperationId() == null) {
-                return;
+            OperationIdentifier operationIdentifier = renderableOutputEvent.getBuildOperationId();
+            if (operationIdentifier == null) {
+                if (rootBuildOperation == null) {
```

Comment:
@breskeby Is this a change in behaviour of what would make it to the console? When will this be null?

Reply:
its not a change in behaviour whats going to the console, but what output is associated with a build operation and by that captured by the build scan plugin. All output from custom created threads used not to have a build operation associated with them. 
the null check is in case anything is logged before _rootBuildOperationStarted_ is called.

Reply:
Thanks